### PR TITLE
[PVR] Context menu cleanup, step 4

### PR DESCRIPTION
**Pull request by ksooo**
Across all PVR windows and dialogs:
- Consequently use same logic (and code) for deciding when to display 'Stop recording' or 'Delete timer'.
- Use same order of menu items

@Jalle19 @xhaggi mind taking a look.

### CODE REVIEW

File: xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp [hunk 1]
```diff
@@ -60,36 +60,26 @@ bool CGUIDialogPVRGuideInfo::ActionStartTimer(const CEpgInfoTagPtr &tag)
 {
   bool bReturn = false;
 
-  if (!tag)
```
Comment (Jalle19) [inline xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp]: This check is now gone, is that intentional?

File: xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp [hunk 2]
```diff
@@ -60,36 +60,26 @@ bool CGUIDialogPVRGuideInfo::ActionStartTimer(const CEpgInfoTagPtr &tag)
...
+    Close();
 
   return bReturn;
 }
 
-bool CGUIDialogPVRGuideInfo::ActionCancelTimer(CFileItemPtr timer)
+bool CGUIDialogPVRGuideInfo::ActionCancelTimer(const CFileItemPtr &timer)
 {
-  bool bReturn(false);
-  if (!timer || !timer->HasPVRTimerInfoTag())
```
Comment (Jalle19) [inline xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp]: Same here

File: xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp (hunk 1 above)
Comment (ksooo) [inline xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp]: Yep. Checked the calling code. Tag is always present.

File: xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp (hunk 2 above)
Comment (ksooo) [inline xbmc/pvr/dialogs/GUIDialogPVRGuideInfo.cpp]: Same answer here. ;-)

**Comment by ksooo** [comment]
Two more fixes for issues reported by @da-anda:
* Prevent delete of read-only timers (like children of repeating timers created by tvheadend)
* Remove context menu item "Add to favourites" for children of repeating timers

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
pvr.vdr.vnsi failed to build on WIN32. Not related to this PR.

**Comment by ksooo** [comment]
If nobody objects I will merge this PR tomorrow.

**Comment by Jalle19** [comment]
Haven't looked at the code in detail but I trust your judgement if you say it works.

**Pull request merged**